### PR TITLE
ci: update GitHub workflows

### PR DESCRIPTION
**Pull request by rflechtner**
## re/ KILTProtocol/ticket#2264

Updates GitHub workflows to no longer use deprecated actions and features which will be retired shortly.

## How to test:

Run as many workflows as possible I guess.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -123,22 +124,20 @@ jobs:
     continue-on-error: ${{ matrix.required == 'optional' }}
 
     steps:
-      - name: Log out node version
-        run: node --version
       - name: Use Node.js 14
         uses: actions/setup-node@v3
         with:
           node-version: 14
```
Comment (ggera) [inline .github/workflows/tests.yml:130]:
```suggestion
          node-version: 16
```

Comment (ggera) [inline .github/workflows/tests.yml:130]:
```suggestion
          node-version: 16
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -175,15 +174,15 @@ jobs:
...
```
Comment (ggera) [inline .github/workflows/tests.yml:176]:
```suggestion
          node-version: 16
```

**Review by ggera — APPROVED** [review]
is the `node-version` upgraded accross the board? if so there are places where it's still 14

**Comment by rflechtner** [comment]
> is the `node-version` upgraded accross the board? if so there are places where it's still 14

I haven't actually updated the node version of our build/test jobs, just the actions that were still using node v12. I'd do that in a separate PR so it's easier to revert if something breaks.

**Pull request merged**